### PR TITLE
Rename _scatter_reduce to scatter_reduce

### PR DESCRIPTION
**Pull request by mikaylagawarecki**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #71256
* __->__ #70750

- Change name from `_scatter_reduce` to `scatter_reduce`
- Make `scatter_reduce` unstructured because gradients not supported for out

Differential Revision: [D33712801](https://our.internmc.facebook.com/intern/diff/D33712801)

### CODE REVIEW

File: aten/src/ATen/native/TensorAdvancedIndexing.cpp [hunk 1]
```diff
@@ -1299,16 +1277,41 @@ TORCH_IMPL_FUNC(scatter_add)
...
- const Tensor& index,
- const c10::string_view reduce,
- const c10::optional<int64_t> output_size,
- const Tensor& out) {
+Tensor scatter_reduce_two_cpu(const Tensor& self,
+                              int64_t dim,
+                              const Tensor& index,
+                              const c10::string_view reduce,
+                              const c10::optional<int64_t> output_size,
+                              const c10::optional<Tensor>& optional_out) {
```
Comment (mikaylagawarecki) [inline aten/src/ATen/native/TensorAdvancedIndexing.cpp:1285]: just want to double check that I didn't misunderstand - by making `scatter_reduce` unstructured and removing the out variant that means we still want the optional_out argument from `torch_scatter` so gradients can be calculated, is that correct? If not I'll remove this additional argument and undo the changes!

Comment (cpuhrsch) [reply aten/src/ATen/native/TensorAdvancedIndexing.cpp:1285]: Let's remove this argument for now and introduce it later on. Our common definition of out variants doesn't support gradients and we shouldn't let this one-off operator intersect with that.

**Review by cpuhrsch — APPROVED** [review]

**Comment by mikaylagawarecki** [comment]
@mikaylagawarecki has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33712801).

**Review by albanD — COMMENTED** [review]
This is missing proper documentation etc as you're making it a public function?

**Comment by cpuhrsch** [comment]
@albanD - docs will land as part of https://github.com/pytorch/pytorch/pull/68580

This PR unblocks that.

**Comment by mikaylagawarecki** [comment]
@mikaylagawarecki has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33712801).

**Comment by albanD** [comment]
Note that the doc update will need to land before the branch cut. Or we will have to remove this function from the public API for the release.

**Pull request closed without merging**